### PR TITLE
Generate JSON and SCSS files from a single source

### PR DESCRIPTION
**Pull request by dcwarwick**
This PR adds the ability to generate the JSON and SCSS files from a single source, as discussed in issue #6, using Handlebars templates and a gulp task.

The source data and templates are placed in a 'source' folder, and the output files ready for client use are placed in a 'dist' folder. This is a common convention, and usefully separates the files for client use from the various configuration and build files that are in the root folder.

Templates are included to produce the two files that already existed. The outputs from these templates are virtually identical to the original files, except for fixing a couple of indentation glitches.
ie,
* ./dist/ibm-colors.scss is virtually identical to the original ./_ibm-colors.scss
* ./dist/ibm-colors.json is virtually identical to the original ./ibm-colors.json
* As we have existing users who may already be pointing SASS/SCSS builds at /path/to/bower_components/ibm-colors/ibm-colors, I have retained the ./_ibm-colors.scss as a simple pass-through to ./dist/ibm-colors.scss, so that our existing users are not broken by the change of folder structure.

The single source of the color definitions is the data structure in ./source/colors.js. If any changes are made here, and the build step run, those changes will be reflected in new versions of the files in ./dist. To run the build step, clone the repo, use ```npm install``` then run ```gulp```.

### CODE REVIEW

File: source/colors.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
+
```
Comment (mbarlock) [inline source/colors.js]: I'm not sure why this file is different than the json block. Or, to put better, I think that that structure is probably all we need, this one seems a bit wordy. Then your generation of the .json file can simply be a `json.stringify()`

**Comment by mbarlock** [comment]
I'm totally fine with using handlebars for the templating, I think it's a bit weird to have the file extension of the template be the one its compiled into. In the example you showed for #6 the had <file-extention>.tpl and I think that could work here. 

The generation of the dist folder should happen at release time and not at checkin. If you can get the gulp tasks then we can update travis to build the dist before it releases the code.

File: source/colors.js (hunk 1 above)
Comment (dcwarwick) [inline source/colors.js]: In Javascript, the order of object fields is not defined. I changed the structure so that the items we would like to be in a particular order (ie, "blue" first then "green" then "teal", etc, and "core" first then "10" then "20" etc within each palette) are represented by arrays, where the order of the elements is significant.

Also, I had thought to include the synonyms ("warm white", "grey", etc) into this structure, although they will not appear in the JSON format of the output -- I haven't done that though as part of this PR.

The template to transform into the JSON is very simple, and using a template rather than just stringify also allows us to set indentation and layout within the JSON file, which isn't significant for mechanical use of the file but can make it look more readable by the human eye.

**Comment by dcwarwick** [comment]
I'd mused over the file names a bit. I originally had an extension, then changed it because by using the final extensions you get edit-time syntax highlighting in most editors, which can be handy. But I'm not sure I like it -- as you say, it's a bit weird -- so I'll reintroduce the extensions. Thanks.

Great point about 'dist' -- I'll remove it from the check-in and gitignore it.

File: source/colors.js (hunk 1 above)
Comment (mbarlock) [inline source/colors.js]: Lovely points. :+1:

File: package.json [hunk 2]
```diff
@@ -15,15 +15,22 @@
     "url": "https://github.com/IBM-Design/colors.git"
   },
   "keywords": [
+    "colors",
     "design",
+    "eyeglass-module",
     "sass",
-    "colors",
-    "eyeglass-module"
+    "scss"
```
Comment (mbarlock) [inline package.json]: :+1: for alphabetizing!

File: ibm-colors.json [hunk 3]
```diff
@@ -1,172 +0,0 @@
-{
```
Comment (mbarlock) [inline ibm-colors.json]: I'm pretty sure people are using this file by creating scss maps with imports. I would like to leave the assets (including the old _ibm-colors.scss) in place and we'll remove them in a 2.0.0 release once we generate the dist at release time.

**Comment by mbarlock** [comment]
Yeah the extension thing with highlighting is a problem. The best solution I could come up with, which I rather like actually, is to put all the significant chunks of other file types in their own appropriately named extension files, then inline them using handlebars

Something like:
```hbs
{{file ./color-functions.scss}}
``

**Comment by dcwarwick** [comment]
Oh yes, I really like the extension files thought -- I've added support for handlebars partials and externalised the color functions exactly as you suggest.

File: ibm-colors.json (hunk 3 above)
Comment (dcwarwick) [inline ibm-colors.json]: Understood. I've reinstated _ibm-colors.scss as before and ibm-colors.json, as literal files NOT generated from the template, for deletion in a future version.

**Comment by mbarlock** [comment]
Sorry this has been sitting so long, you're updates slipped passed me. This looks great!

**Comment by kevinSuttle** [comment]
Does this close https://github.com/IBM-Design/colors/issues/6?

**Comment by mbarlock** [comment]
I believe so
